src/expr/node_manager.cpp

```diff
@@ -248,23 +240,17 @@ NodeManager::~NodeManager() {
   d_options = NULL;
 }
 
-unsigned NodeManager::registerDatatype(Datatype* dt) {
-  unsigned sz = d_ownedDatatypes.size();
-  d_ownedDatatypes.push_back( dt );
+unsigned NodeManager::registerDatatype(std::shared_ptr<DType> dt)
```

Comment:
Since we are anyway modifying this function, can we switch to `size_t` instead of `unsigned`?